### DISCUSSION TITLE
DBML generator for Prisma

### DISCUSSION THREAD

**Discussion by giscus** [notiz-dev/notiz#211] [General]
# DBML generator for Prisma

Visualize Prisma Schema as Entity-Relationship Diagram

https://notiz.dev/blog/prisma-dbml-generator

**Comment by rui494274693** [comment]
Table User {
  id Int [pk, increment]
  createdAt DateTime [default: `now()`, not null]
  updatedAt DateTime [not null]
  email String [unique, not null]
  name String
  posts Post
  profile Profile
  role Role [not null, default: 'USER', note: 'user role']
}

Table Profile {
  id Int [pk, increment]
  bio String
  user User [not null]
  userId Int [unique, not null]

  Note: 'User profile'
}

Table Post {
  id Int [pk, increment]
  title String [not null, default: '']
  content String
  published Boolean [not null, default: false]
  author User
  authorId Int
  categories Category
}

Table Category {
  id Int [pk, increment]
  name String [not null]
  posts Post
}

Table CategoryToPost {
  categoryId Int [ref: > Category.id]
  postId Int [ref: > Post.id]
}

Enum Role {
  ADMIN
  USER
}

Ref: Profile.userId - User.id

Ref: Post.authorId > User.id

**Comment by mash180sx** [comment]
Does this support Prisma's multi-folder schema?

**Comment by zahidshowrav** [comment]
$ pnpm dlx prisma generate
```
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Error: Generator "prisma-dbml-generator" failed:

/bin/sh: 1: prisma-dbml-generator: not found
```
